lib/ASRouter.jsm

```diff
@@ -1480,10 +1562,12 @@ class _ASRouter {
   }
 
   async addImpression(message) {
-    const provider = this.state.providers.find(p => p.id === message.provider);
+    const groupsWithFrequency = this.state.groups.filter(
```

Comment:
This is good because it means we can use groups just for prefs if we want (like in the case of CFR features v.s. addons) without adding any extra unnecessary impression data.